File: programmers/shim/Python/level1/20220525_체육복.py
def solution(n, lost, reserve):
    l=[0]*n
    for i in lost:
        l[i-1]-=1
    for i in reserve:
        l[i-1]+=1
    for i in range(len(l)):
        if l[i]==-1:
            if(i!=0 and l[i-1]>0):
                l[i]+=1
                l[i-1]-=1
            elif(i!=len(l)-1 and l[i+1]>0):
                l[i]+=1
                l[i+1]-=1
    answer = sum(1 for i in l if i>=0)
    return answer

File: programmers/shim/Java/level1/20220525_체육복.java
package Study_2022.programmers.shim.Java.level1;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;
        int s[]=new int[n];
        for(int i=0; i<lost.length; i++)
            s[lost[i]-1]--;
        for(int i=0; i<reserve.length; i++)
            s[reserve[i]-1]++;
        for(int i=0; i<n; i++){
            if(s[i]==-1){
                if(i!=0&&s[i-1]>0){
                    s[i]++;
                    s[i-1]--;
                }else if(i!=s.length-1&&s[i+1]>0){
                    s[i]++;
                    s[i+1]--;
                }
            }
        }
        for(int i=0; i<n; i++)
            if(s[i]>=0) answer++;
        return answer;
    }
}
